Deploy task fails with Gradle 6.2.1 on statsd artifact

Since upgrading the build to use Gradle 6.2.1 in #1857 the deploy job is failing, specifically for the statsd artifact, with the following error. For instance, see https://circleci.com/gh/micrometer-metrics/micrometer/5525

```
Execution failed for task ':artifactoryDeploy'.
> java.util.concurrent.ExecutionException: java.lang.IllegalStateException: Artifact micrometer-registry-statsd-1.4.0-SNAPSHOT.jar wasn't produced by this build.
```

This leads me to believe it has to do with the shadow plugin we are using with the statsd module.

@anuraaga For your information, a similar failure can be reproduced with the following command:

```
$ ./gradlew clean pTML
...
> Task :micrometer-registry-statsd:publishNebulaPublicationToMavenLocal FAILED

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':micrometer-registry-statsd:publishNebulaPublicationToMavenLocal'.
> Failed to publish publication 'nebula' to repository 'mavenLocal'
   > Artifact micrometer-registry-statsd-1.4.0-SNAPSHOT.jar wasn't produced by this build.
...
$
```

--- a/implementations/micrometer-registry-statsd/build.gradle
+++ b/implementations/micrometer-registry-statsd/build.gradle
@@ -37,6 +37,8 @@ jar.dependsOn shadowJar
 publishing {
     publications {
         withType(MavenPublication) {
+            artifact(shadowJar)
+
             pom.withXml {
                 asNode()
                         .dependencies

